src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusServiceTest.java

```diff
@@ -298,6 +299,96 @@ public void should_return_invalid_parameter_exception_for_dcn_less_than_6_chars(
             .hasMessageMatching("DCN number has to be at least 6 characters long");
     }
 
+    @Test
+    public void should_return_envelopes_and_events_from_db_when_ccd_id_is_given() {
+        // given
+        List<ProcessEvent> events = asList(
+            event(Event.DOC_UPLOADED, "A", now(), null),
+            event(Event.DOC_PROCESSED_NOTIFICATION_SENT, "A", now().plusSeconds(1), null),
+            event(Event.DOC_FAILURE, "B", now().minusSeconds(2), "reason2")
+        );
+        String ccdId = "7452652475246754";
+        List<Envelope> envelopes = asList(envelope("A"), envelope("B"));
+        given(envelopes.get(0).getZipFileName()).willReturn("hello.zip");
+        given(envelopeRepo.findByCcdId(ccdId)).willReturn(envelopes);
+        given(eventRepo.findByZipFileName("hello.zip")).willReturn(events);
+        // when
+        ZipFileStatus result = service.getStatusByCcdId(ccdId);
+        // then
+        assertThat(result.ccdId).isEqualTo(ccdId);
+        assertThat(result.fileName).isEqualTo("hello.zip");
```

Comment:
I think this should be null (not exist) if we are returning what is searched for